homeassistant/components/juicenet.py

```diff
@@ -0,0 +1,104 @@
+"""
+Support for Juicenet cloud.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/juicenet
+"""
+
+import logging
+import time
+
+import voluptuous as vol
+
+from homeassistant.helpers import discovery
+from homeassistant.const import CONF_ACCESS_TOKEN
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['python-juicenet==0.0.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'juicenet'
+
+SERVICE_ADD_NEW_DEVICES = 'add_new_devices'
+SERVICE_REFRESH_STATES = 'refresh_state_from_juicenet'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_ACCESS_TOKEN): cv.string
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up the Juicenet component."""
+    import pyjuicenet
+
+    hass.data[DOMAIN] = {}
+    hass.data[DOMAIN]['unique_ids'] = []
+    hass.data[DOMAIN]['entities'] = {}
+
+    access_token = config[DOMAIN].get(CONF_ACCESS_TOKEN)
+
+    hass.data[DOMAIN]['api'] = pyjuicenet.Api(access_token)
+
+    def force_update(call):
+        """Force all devices to poll the Juicenet API."""
+        _LOGGER.info("Refreshing Juicenet states from API")
+        for entity_list in hass[DOMAIN]['entities'].values():
+            for entity in entity_list:
+                time.sleep(1)
+                entity.schedule_update_ha_state(True)
+    hass.services.register(DOMAIN, SERVICE_REFRESH_STATES, force_update)
+
+    def pull_new_devices(call):
+        """Pull new devices added to users Juicenet account since startup."""
+        _LOGGER.info("Getting new devices from Juicenet API")
+        discovery.load_platform(hass, 'sensor', DOMAIN, {}, config)
```

Comment:
This is a very weird way of "pulling" new devices. Why can't you talk to the API directly?